README.md

```diff
@@ -11,47 +11,116 @@ ADaM in R Asset Library
+
+We have four priority design principles to achieve the main goal as follows:
+
+_**Usability**_
+
+All `{admiral}` functions should be easy to use. 
+* Our documentation is an absolute priority. Each function reference page should cover the purpose,
+descriptions of each argument with permitted values, the expected input and output, with clear real-life
+examples - so that users don’t need to dig through code to find answers.
+* Vignettes then complement the functional documentation to help users see how best they can be applied to
+achieve customary ADaM requirements.
+* Functions should be written and structured in a way that users are able to read, re-use or extend them
+for study specific purposes if needed (see Readability below).
+
+_**Simplicity**_
+
+Each function has a clear purpose.
+* We try not to ever design single functions that could achieve numerous very different derivations, as for
+example if you as a user pick up a function with >10 different arguments then chances are it is going to be
+difficult to understand if this function could be applied for your specific need. The intention is that
+arguments/parameters can influence how the output of a function is calculated, but not change the purpose of
+the function.
+* On the other hand, we do try to combine similar tasks and algorithms into one function to reduce the amount
+of repetitive, similar functionality (e.g. one study day calculation rather than a function per variable).
+Importantly here though, functions cannot be too general and trying to fulfil multiple, complex purposes.
+Similarly, functions should not allow expressions as arguments that are used as code snippets in function calls.
+* We recommend to avoid copy and paste of complex computational algorithms or repetitive code like checks
+and advise to wrap them into a function. However we would also like to avoid multi-layered functional nesting,
+so this needs to be considered carefully to keep the nesting of 3-4 functions an exception rather than the rule.
+
+_**Findability**_
+
+In a growing codebase, across a family of packages, we make every effort to make our functions easily findable.
+* We use consistent naming conventions across all our functions, and provide vignettes and ADaM templates that
+help users to get started and build familiarity. Each `{admiral}` family package website is searchable.
+* We avoid repetitive functions that will do similar tasks (as explained above with study day example).
+* Each package extension is kept focused on the specific scope, i.e. any common need that could be raised up
+to the core package will be.
+
+_**Readability**_
+
+We have a very prescriptive Programming Strategy guide that all our developers and contributors must follow,
+so that all our code has a high degree of consistency and readability.
+* We mandate use of tidyverse (e.g. dplyr) over similar functionality existing in base R.
```

Comment:
Do we mandate this?  I thought we mandate tidyverse style guide?  I don't see this in the Programming Strategy Guide

Reply:
@thomas-neitmann i remember you asked for this. should i add to programming strategy too?

Reply:
@bms63 do you think it needs a broader discussion? I think this is basically the direction we going, it just needs to be added to the programming strategy.

Reply:
I'm fine with that we should just state that somewhere.  As this is the first time I've seen this in writing.

Reply:
I added it to the programming strategy too.